source/m2mnsdlinterface.cpp

```diff
@@ -1649,6 +1641,25 @@ void M2MNsdlInterface::handle_bootstrap_error()
     _observer.bootstrap_error();
 }
 
+void M2MNsdlInterface::set_connection_handler(M2MConnectionHandler *connection_handler)
```

Comment:
We dont need to worry about backward compatibility of M2MNsdlInterface class, its our internal class , we can modify the parameters the way we want, 
Tero has a good point, so if @kuggenhoffen can fix it, we can merge this PR in.